### PR TITLE
week assessment 2

### PR DESCRIPTION
**Pull request by GreatScott-Ant**
# Submitting Week 2 Assessment

### ✅ Completed Files

Select the following as confirmation that the files in this repository are complete.

- [ ] Career Readiness
- [ ] Coding Challenges
- [ ] Coding Critique
- [ ] Coding Concepts
- [ ] Interview Questions

### ✋ Reminder

Do NOT merge your assessment!

### 📝 Feedback

If you would like specific feedback please leave details here:

### CODE REVIEW

**Comment by GreatScott-Ant** [comment]
Pseudo code critique/advice always welcome, still working on building vocabulary understanding and usage. (Honestly anything that is going to push me to evolve as a developer!)

File: code-challenges.test.js [hunk 1]
```diff
@@ -1,51 +1,160 @@
...
+
+// const numbersArray1 = [6, 7, 8, 9, 10]
+// // Expected output: [18, 21, 24, 27, 30]
+// const numbersArray2 = [24, 27, 30, 33, 36]
+// // Expected output: [72, 81, 90, 99, 108]
+
+// expect(multiplyByThree(numbersArray1)).toEqual([18, 21, 24, 27, 30])
+// expect(multiplyByThree(numbersArray2)).toEqual([72, 81, 90, 99, 108])
+// })
+// })
```
Comment (chrisyhlee) [inline code-challenges.test.js:28]: Pay attention to your indentation. This code is very hard to read, which can be easily fixed with proper indentation!

File: code-challenges.test.js [hunk 2]
```diff
@@ -1,51 +1,160 @@
...
+//         expect(diviByThree(object1)).toEqual(15 + " " + 'is divisible by three')
+//         // Expected output: "15 is divisible by three"
+
+//         const object2 = {number: 0}
+//         expect(diviByThree(object2)).toEqual(0 + " " + 'is divisible by three')
+//         // Expected output: "0 is divisible by three"
+
+//         const object3 = {number: -7}
+        // expect(diviByThree(object3)).toEqual(-7 + " " + 'is not divisible by three')
+//         // Expected output: "-7 is not divisible by three"
```
Comment (chrisyhlee) [inline code-challenges.test.js:69]: In your expect, you want exactly what is given in the expected output. You can just copy and paste the expected output into the parentheses after toEqual

File: code-challenges.test.js [hunk 3]
```diff
@@ -1,51 +1,160 @@
...
+// Pseudo code:
+
+//Input: an array of strings
+//Output: same array of strings but with letter at 0 index capitalized
+
+const capSized = (array) => {
+
+    //return an array as a string 
+    return array.map((nouns) => {
+        //return string that takes 0 index position, capitolizes it, then combines with other characters to become array again
```
Comment (chrisyhlee) [inline code-challenges.test.js:145]: Not become an array again, it is combining the first letter capitalized with the rest of the word, so it is still a string.

File: code-critique.js [hunk 4]
```diff
@@ -6,5 +6,56 @@
...
\ No newline at end of file
+// Things to consider in your critique: Were you surprised by the output? Was the output what you expected? Is the correct output being rendered? What do you like about the code? What do you dislike? Are there best practices that are not being followed? Would you take a different approach? What did you learn? 
+
+//When I initally pasted the prompt into ChatGPT, I did not include the coding language, so I believe it gave me the response in Python. I went back and 
+//included "Javascript", and received the latter code output. It was more of what I expected and was not too surprised by the results. I found it intersting
+//that for the declarative statement nothing was chosen (var, let, const); instead it just said "function". What confused me was the placement of the return at the end of the code and after the first call
+//for console.log. I was also surprised how it configured the console.log for if there weren't any vowels to output a string stating that. Overall the structure and method used is the general approach I 
+//thought to take: creating a variable that listed the vowels, including the vowels capitalized as well, using a for/let statement.
```
Comment (chrisyhlee) [inline code-critique.js:61]: It's important to understand that AI can be a great tool, but it needs to be used properly. Use it only to expand your own learning, and not just to take the answer it gives at face value as gospel.

File: coding-concepts.js [hunk 5]
```diff
@@ -21,16 +21,16 @@ const greeter = (name) => {
 }
 // console.log(greeter("LEARN Student"))
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: Hello, LEARN Student
+// b) Verify and explain: undefined; there was no variable statement defining what name was, so name argument is still undefined.
```
Comment (chrisyhlee) [inline coding-concepts.js:25]: The function has no return, so we get undefined

**Review by chrisyhlee — APPROVED** [review]
You have a good foundational understanding of these concepts, and I'm glad to see you are able to explain the code in your pseudo. Make sure you pay attention to your indentation when writing your code. Also, I suggest you work on your interview question answers, as they can be much more elaborate. You don't want the interviewer to doubt your understanding of the concepts. Overall this assessment displays Novice.

**Pull request merged**